### PR TITLE
chore(Typescript): Update Typescript declarations and add documentation

### PR DESCRIPTION
**Pull request by hughsw**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -583,6 +583,25 @@ process.on('uncaughtException', async () => {
   process.exit(1);
 });
 ```
+#### Queue#isRunning()
+
+Returns `true` unless the Queue is shutting down due to a call to `Queue#close()`;
```
Comment (skeggse) [inline README.md]: :)
```suggestion
Returns `true` unless the Queue is shutting down due to a call to `Queue#close()`.
```

File: README.md [hunk 2]
```diff
@@ -583,6 +583,25 @@ process.on('uncaughtException', async () => {
...
+
+#### Queue#ready([cb])
+
+Promise resolves to the queue (or callback is called wth `null` argument) when the queue (and Redis) are ready for jobs.  Learn more about `'ready'` in [Queue Local Events](#queue-local-events).
+
+```js
+const Queue = require('bee-queue');
+const queue = new Queue('example');
+queue.ready()
+  .then(async queue => {
```
Comment (skeggse) [inline README.md]: for consistency
```suggestion
  .then(async (queue) => {
```

File: README.md [hunk 3]
```diff
@@ -583,6 +583,25 @@ process.on('uncaughtException', async () => {
...
+    console.log('isRunning:', queue.isRunning());
+    const checkHealth = await queue.checkHealth();
+    console.log('checkHealth:', JSON.stringify(checkHealth));
```
Comment (skeggse) [inline README.md]: not necessary for either real use or example comprehension
```suggestion
    console.log('checkHealth:', checkHealth);
```

File: README.md [hunk 4]
```diff
@@ -583,6 +583,25 @@ process.on('uncaughtException', async () => {
...
+  })
+  .catch(err => console.log('unreadyable', err));
```
Comment (skeggse) [inline README.md]: for consistency
```suggestion
  .catch((err) => console.log('unreadyable', err));
```

File: index.d.ts [hunk 5]
```diff
@@ -45,6 +45,11 @@ declare class BeeQueue {
   close(timeout?: number | null): Promise<void>;
   close(timeout: number | undefined | null, cb: () => void): void;
 
+  isRunning(): boolean;
+
+  ready(): Promise<this>;
+  ready(cb: () => void): void;
```
Comment (skeggse) [inline index.d.ts]: If I'm not mistaken, this still returns a `Promise` - maybe these two definitions should just get merged to have an optional `cb`?
```suggestion
  ready(cb: () => void): Promise<this>;
```

Comment (hughsw) [reply index.d.ts]: Thanks -- yes, a promise is returned.

Regarding merging the declarations: I'm new to the creating of `*.d.ts` files.  However, this multi-line pattern appears throughout the `@tkypes/node/*.d.ts` files, e.g. in `fs.d.ts` which has a lot of legacy support...

File: README.md (hunk 2 above)
Comment (hughsw) [reply README.md]: Stylistic habit from my current long-term project...

**Pull request merged**